sql/src/main/java/org/apache/druid/sql/calcite/planner/SqlStatementHandler.java

```diff
@@ -43,7 +44,7 @@
   PrepareResult prepareResult();
   PlannerResult plan() throws ValidationException;
   String statementKind();
-  SqlNode targetDataSource();
+  @Nullable SqlNode targetDataSource();
```

Comment:
nit: would add a short javadoc, explaining that it is null of statement kind is SELECT / non dml.

Reply:
I've added a comment to the specific select implementation that returns null but can make that as a javadoc: https://github.com/apache/druid/pull/14074/files#diff-483e089932b8faa5ca9669607eab7fcba883d27e2899a7c9fba2f37afb9de97bR241

I